Core/Core/SwiftUIViews/Tag.swift

```diff
@@ -0,0 +1,96 @@
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <https://www.gnu.org/licenses/>.
+//
+
+import SwiftUI
+
+@available(iOS 13.0, *)
+public enum Tag {
+    /// Wrapper around LocalizedStringKey that provides access to the string
+    public struct LocalizedString: ExpressibleByStringInterpolation {
+        public let key: LocalizedStringKey
+        public let string: String
+        public init(stringLiteral value: String) {
+            key = LocalizedStringKey(value)
+            string = value
+        }
+        public init(stringInterpolation: StringInterpolation) {
+            key = LocalizedStringKey(stringInterpolation: stringInterpolation.keyInterpolation)
+            string = stringInterpolation.components.joined()
+        }
+        public struct StringInterpolation: StringInterpolationProtocol {
+            public var keyInterpolation: LocalizedStringKey.StringInterpolation
+            public var components: [String] = []
+
+            public init(literalCapacity: Int, interpolationCount: Int) {
+                keyInterpolation = .init(literalCapacity: literalCapacity, interpolationCount: interpolationCount)
+                components.reserveCapacity(literalCapacity + interpolationCount)
+            }
+            public mutating func appendLiteral(_ literal: String) {
+                keyInterpolation.appendLiteral(literal)
+                components.append(literal)
+            }
+            public mutating func appendInterpolation(_ string: String) {
+                keyInterpolation.appendInterpolation(string)
+                components.append(string)
+            }
+            public mutating func appendInterpolation<Subject: ReferenceConvertible>(_ subject: Subject, formatter: Formatter? = nil) {
+                keyInterpolation.appendInterpolation(subject, formatter: formatter)
+                components.append(formatter?.string(for: subject as? NSObject) ?? "\(subject)")
+            }
+            public mutating func appendInterpolation<Subject: NSObject>(_ subject: Subject, formatter: Formatter? = nil) {
+                keyInterpolation.appendInterpolation(subject, formatter: formatter)
+                components.append(formatter?.string(for: subject) ?? "\(subject)")
+            }
+            public mutating func appendInterpolation<T: _FormatSpecifiable>(_ value: T) {
```

Comment:
It's gross, but these psuedo-private "_Type"s are part of the SwiftUI public interface, and have to be dealt with.